modules/deployment-model/src/test/java/org/mule/runtime/artifact/activation/internal/TrackingArtifactClassLoaderResolverDecoratorTestCase.java

```diff
@@ -80,7 +80,8 @@ public void setUp() throws Exception {
         .thenReturn(applicationClassLoader2);
     when(artifactClassLoaderResolver.createApplicationClassLoader(any(), any(Supplier.class)))
         .thenReturn(applicationClassLoader3);
-    when(artifactClassLoaderResolver.createApplicationClassLoader(any(), any(), any())).thenReturn(applicationClassLoader4);
```

Comment:
The checks for the 2 new methods is missing.

Reply:
It's not an old implementation, it's the one used in the Runtime. The new methods (and some of the older ones too) are not being used in the Runtime, but this implementation should do what it says it does, why restricting the usage of the new methods?

Reply:
Because it will be only used in the framework